app/src/main/java/com/epfl/drawyourpath/notifications/AlarmReceiver.kt

```diff
@@ -0,0 +1,74 @@
+package com.epfl.drawyourpath.notifications
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.login.ENABLE_ONETAP_SIGNIN
+import com.epfl.drawyourpath.login.LoginActivity
+import com.epfl.drawyourpath.login.RESTORE_USER_IN_KEYCHAIN
+import com.epfl.drawyourpath.mainpage.USE_MOCK_CHALLENGE_REMINDER
+
+/**
+ * Class that can execute code when it receives an alarm (with onReceive())
+ */
+class AlarmReceiver : BroadcastReceiver() {
+
+    /**
+     * Sends a notification when an alarm is received
+     */
+    override fun onReceive(context: Context, intent: Intent) {
+        val notificationManager = ContextCompat.getSystemService(
+            context,
+            NotificationManager::class.java
+        ) as NotificationManager
+
+        val disableAutoReconnection = intent.getBooleanExtra(USE_MOCK_CHALLENGE_REMINDER, false)
+
+        notificationManager.sendChallengeReminderNotification(context, !disableAutoReconnection)
+        // Send other time-based notifications here
+    }
+}
+
+/**
+ * Sends a challenge reminder notification
+ * @param applicationContext: the application context
+ */
+fun NotificationManager.sendChallengeReminderNotification(
+    applicationContext: Context, enableAutoReconnection: Boolean
+) {
```

Comment:
This function could be split so that parts of it can be reused by other notifications.

Reply:
I can split this function in smaller ones to be more readable, this is for sure a good idea, but I do not see how to make reusable functions out of it, as most (if not all) the code is specific for a challenge reminder notification and not for notifications in general. I still modularized it a bit, for readability mainly, thanks for pointing out this function !